### ISSUE TITLE
SUSE Linux Backup is failing ERROR: Could not create ISO image (with /usr/bin/mkisofs)

### ISSUE THREAD

**Issue by parasharsingh** [rear/rear#804]
* rear version (/usr/sbin/rear -V):
* OS version (cat /etc/rear/os.conf or lsb_release  -a):
* rear configuration files (cat /etc/rear/site.conf or cat /etc/rear/local.conf):
* Brief description of the issue
* Work-around, if any

````
uswix163# rear -V
Relax-and-Recover 1.18 / Git
uswix163# cat /etc/rear/os.conf
OS_VENDOR=SUSE_LINUX
OS_VERSION=11
uswix163# cat /etc/rear/local.conf
# Default is to create Relax-and-Recover rescue media as ISO image
# set OUTPUT to change that
# set BACKUP to activate an automated (backup and) restore of your data
# Possible configuration values can be found in /usr/share/rear/conf/default.conf
#
# This file (local.conf) is intended for manual configuration. For configuration
# through packages and other automated means we recommend creating a new
# file named site.conf next to this file and to leave the local.conf as it is.
# Our packages will never ship with a site.conf.
OUTPUT=ISO
BACKUP=NETFS
BACKUP_URL="nfs://10.20.0.35/kowig04_pkg/Shared6/Pkg/rear"
BACKUP_OPTIONS="nfsvers=3,nolock"
EXCLUDE_MOUNTPOINTS=( "/sapmnt" )

While trying to create a Backup its terminating with following error message.

uswix163#
uswix163# rear -d -D mkbackup
Relax-and-Recover 1.18 / Git
Using log file: /var/log/rear/rear-uswix163.log
mkdir: created directory `/var/lib/rear/output'
Using UEFI Boot Loader for Linux (USING_UEFI_BOOTLOADER=1)
Creating disk layout
Creating root filesystem layout
Copying files and directories
Copying binaries and libraries
Copying kernel modules
Creating initramfs
Making ISO image
ERROR: Could not create ISO image (with /usr/bin/mkisofs)
Aborting due to an error, check /var/log/rear/rear-uswix163.log for details
You should also rm -Rf /tmp/rear.3EY7Ue3zJBnJ4RX
Terminated
````

**Comment by gdha** [comment]
Please install `ebiso` if UEFI booting is required - download from http://download.opensuse.org/repositories/Archiving:/Backup:/Rear/SLE_11/x86_64/

**Comment by parasharsingh** [comment]
Even after installing rpm I'm getting same error:

uswix162# rpm -ivh ebiso-0.1.4-1.x86_64.rpm
warning: ebiso-0.1.4-1.x86_64.rpm: Header V3 DSA signature: NOKEY, key ID 725a0c43
Preparing...                ########################################### [100%]
   1:ebiso                  ########################################### [100%]
uswix162# rear -d -D mkbackup
Relax-and-Recover 1.16 / Git
Using log file: /var/log/rear/rear-uswix162.log
Using UEFI Boot Loader for Linux (USING_UEFI_BOOTLOADER=1)
Creating disk layout
Creating root filesystem layout
Copying files and directories
Copying binaries and libraries
Copying kernel modules
Creating initramfs
Making ISO image
ERROR: Could not create ISO image (with /usr/bin/mkisofs)
Aborting due to an error, check /var/log/rear/rear-uswix162.log for details
You should also rm -Rf /tmp/rear.VT25b9gac3pXGae
Terminated
uswix162#

**Comment by gozora** [comment]
Hi,
ReaR configuration needs to be adapted as well.
Please check: http://www.it3.be/2015/10/27/uefi-iso-boot-with-ebiso/

**Comment by parasharsingh** [comment]
under log file I can see following:
----------------------------snipped-----------------------------------
+++ date '+%Y-%m-%d %H:%M:%S.%N '
++ echo '2016-03-23 11:56:33.273457525 Including ISO UEFI boot (as triggered by USING_UEFI_BOOTLOADER=1)'
2016-03-23 11:56:33.273457525 Including ISO UEFI boot (as triggered by USING_UEFI_BOOTLOADER=1)
++ pushd /tmp/rear.VT25b9gac3pXGae/tmp/isofs
++ /usr/bin/mkisofs -v -o /var/lib/rear/output/rear-uswix162.iso -b isolinux/isolinux.bin -c isolinux/boot.cat -
no-emul-boot -boot-load-size 4 -boot-info-table -R -J -volid RELAXRECOVER -eltorito-alt-boot -e boot/efiboot.img
 -no-emul-boot -v -iso-level 3 .
genisoimage: option '-e' is ambiguous
Usage: genisoimage [options] -o file directory ...

Use genisoimage -help
to get a list of valid options.

Report problems to debburn-devel@lists.alioth.debian.org.
++ StopIfError 'Could not create ISO image (with /usr/bin/mkisofs)'
++ ((  1 != 0  ))
++ Error 'Could not create ISO image (with /usr/bin/mkisofs)'
++ '[' Could not create ISO image '(with' '/usr/bin/mkisofs)' -eq Could not create ISO image '(with' '/usr/bin/m
kisofs)' ']'
++ EXIT_CODE=1
++ VERBOSE=1
++ LogPrint 'ERROR: Could not create ISO image (with /usr/bin/mkisofs)'
++ Log 'ERROR: Could not create ISO image (with /usr/bin/mkisofs)'
++ test 1 -gt 0
+++ Stamp
+++ date '+%Y-%m-%d %H:%M:%S.%N '
++ echo '2016-03-23 11:56:33.277361395 ERROR: Could not create ISO image (with /usr/bin/mkisofs)'
2016-03-23 11:56:33.277361395 ERROR: Could not create ISO image (with /usr/bin/mkisofs)
++ Print 'ERROR: Could not create ISO image (with /usr/bin/mkisofs)'
++ test 1
++ echo -e 'ERROR: Could not create ISO image (with /usr/bin/mkisofs)'
++ has_binary caller
++ for bin in '$@'
++ type caller
++ return 0
++ echo '=== Stack trace ==='
=== Stack trace ===
++ local c=0
++ caller 0
------------------snipped-----------------------------

**Comment by parasharsingh** [comment]
I made the changes in configuration file as well but still this is not working.

uswix162# cat /etc/rear/local.conf
# Default is to create Relax-and-Recover rescue media as ISO image
# set OUTPUT to change that
# set BACKUP to activate an automated (backup and) restore of your data
# Possible configuration values can be found in /usr/share/rear/conf/default.conf
#
# This file (local.conf) is intended for manual configuration. For configuration
# through packages and other automated means we recommend creating a new
# file named site.conf next to this file and to leave the local.conf as it is.
# Our packages will never ship with a site.conf.
OUTPUT=ISO
BACKUP=NETFS
BACKUP_URL="nfs://10.20.0.35/kowig04_pkg/Shared6/Pkg/rear"
BACKUP_OPTIONS="nfsvers=3,nolock"
EXCLUDE_MOUNTPOINTS=( "/sapmnt" )
ISO_MKISOFS_BIN=/usr/bin/ebiso
uswix162#
uswix162# rear -d -D mkbackup
Relax-and-Recover 1.16 / Git
Using log file: /var/log/rear/rear-uswix162.log
Using UEFI Boot Loader for Linux (USING_UEFI_BOOTLOADER=1)
Creating disk layout
Creating root filesystem layout
Copying files and directories
Copying binaries and libraries
Copying kernel modules
Creating initramfs
Making ISO image
Wrote ISO image: /var/lib/rear/output/rear-uswix162.iso ()
Copying resulting files to nfs location
ERROR: Could not copy files to nfs location
Aborting due to an error, check /var/log/rear/rear-uswix162.log for details
You should also rm -Rf /tmp/rear.yQCQmuw7D95ysgf
Terminated


This is what I see in log file:

--------Snipped---------------------
+++ date '+%Y-%m-%d %H:%M:%S.%N '
++ echo '2016-03-23 12:44:34.875993443 Including ISO UEFI boot (as triggered by USING_UEFI_BOOTLOADER=1)'
2016-03-23 12:44:34.875993443 Including ISO UEFI boot (as triggered by USING_UEFI_BOOTLOADER=1)
++ pushd /tmp/rear.yQCQmuw7D95ysgf/tmp/isofs
++ /usr/bin/ebiso -v -o /var/lib/rear/output/rear-uswix162.iso -b isolinux/isolinux.bin -c isolinux/boot.cat -no-emul-boot -boot-load-size 4 -boot-info-table -R -J -volid RELAXRECOVER -eltorito-alt-boot -e boot/efiboot.img -no-emul-boot -v -iso-level 3 .
++ StopIfError 'Could not create ISO image (with /usr/bin/ebiso)'
++ ((  0 != 0  ))
++ popd
++ ISO_IMAGES=("${ISO_IMAGES[@]}" "$ISO_DIR/$ISO_PREFIX.iso")
++ iso_image_size=($(du -h "$ISO_DIR/$ISO_PREFIX.iso"))
+++ du -h /var/lib/rear/output/rear-uswix162.iso
du: cannot access `/var/lib/rear/output/rear-uswix162.iso': No such file or directory
++ LogPrint 'Wrote ISO image: /var/lib/rear/output/rear-uswix162.iso ()'
++ Log 'Wrote ISO image: /var/lib/rear/output/rear-uswix162.iso ()'
++ test 1 -gt 0
+++ Stamp
+++ date '+%Y-%m-%d %H:%M:%S.%N '
++ echo '2016-03-23 12:44:34.882682980 Wrote ISO image: /var/lib/rear/output/rear-uswix162.iso ()'
2016-03-23 12:44:34.882682980 Wrote ISO image: /var/lib/rear/output/rear-uswix162.iso ()
++ Print 'Wrote ISO image: /var/lib/rear/output/rear-uswix162.iso ()'
++ test 1
++ echo -e 'Wrote ISO image: /var/lib/rear/output/rear-uswix162.iso ()'
++ RESULT_FILES=("${RESULT_FILES[@]}" "$ISO_DIR/$ISO_PREFIX.iso")
+ test 1
+ set +x
2016-03-23 12:44:34.885994423 Including output/ISO/Linux-i386/85_check_for_errors.sh
+ . /usr/share/rear/output/ISO/Linux-i386/85_check_for_errors.sh
++ [[ 1 -eq 1 ]]
++ return
+ test 1
+ set +x
2016-03-23 12:44:34.888637760 Including output/default/94_grub_rescue.sh
+ . /usr/share/rear/output/default/94_grub_rescue.sh
++ [[ ! n =~ ^[yY1] ]]
++ return
+ test 1
+ set +x
2016-03-23 12:44:34.891913899 Including output/default/95_copy_result_files.sh
+ . /usr/share/rear/output/default/95_copy_result_files.sh
+++ url_scheme nfs://10.20.0.35/kowig04_pkg/Shared6/Pkg/rear
+++ local url=nfs://10.20.0.35/kowig04_pkg/Shared6/Pkg/rear
+++ local scheme=nfs
+++ echo nfs
+++ grep -q :
+++ echo nfs
++ local scheme=nfs
+++ url_host nfs://10.20.0.35/kowig04_pkg/Shared6/Pkg/rear
+++ local url=nfs://10.20.0.35/kowig04_pkg/Shared6/Pkg/rear
+++ local host=10.20.0.35/kowig04_pkg/Shared6/Pkg/rear
+++ echo 10.20.0.35
++ local host=10.20.0.35
+++ url_path nfs://10.20.0.35/kowig04_pkg/Shared6/Pkg/rear
+++ local url=nfs://10.20.0.35/kowig04_pkg/Shared6/Pkg/rear
+++ local path=10.20.0.35/kowig04_pkg/Shared6/Pkg/rear
+++ echo /kowig04_pkg/Shared6/Pkg/rear
++ local path=/kowig04_pkg/Shared6/Pkg/rear
+++ output_path nfs /kowig04_pkg/Shared6/Pkg/rear
+++ local scheme=nfs
+++ local path=/kowig04_pkg/Shared6/Pkg/rear
+++ case $scheme in
+++ path=/tmp/rear.yQCQmuw7D95ysgf/outputfs/uswix162
+++ echo /tmp/rear.yQCQmuw7D95ysgf/outputfs/uswix162
++ local opath=/tmp/rear.yQCQmuw7D95ysgf/outputfs/uswix162
++ [[ -z /tmp/rear.yQCQmuw7D95ysgf/outputfs/uswix162 ]]
++ [[ -z nfs://10.20.0.35/kowig04_pkg/Shared6/Pkg/rear ]]
++ [[ nfs == \o\b\d\r ]]
++ [[ nfs == \t\a\p\e ]]
++ LogPrint 'Copying resulting files to nfs location'
++ Log 'Copying resulting files to nfs location'
++ test 1 -gt 0
+++ Stamp
+++ date '+%Y-%m-%d %H:%M:%S.%N '
++ echo '2016-03-23 12:44:34.900658732 Copying resulting files to nfs location'
2016-03-23 12:44:34.900658732 Copying resulting files to nfs location
++ Print 'Copying resulting files to nfs location'
++ test 1
++ echo -e 'Copying resulting files to nfs location'
++ case "$scheme" in
++ test /var/lib/rear/output/rear-uswix162.iso
++ Log 'Copying files '\''/var/lib/rear/output/rear-uswix162.iso'\'' to nfs location'
++ test 1 -gt 0
+++ Stamp
+++ date '+%Y-%m-%d %H:%M:%S.%N '
++ echo '2016-03-23 12:44:34.902757177 Copying files '\''/var/lib/rear/output/rear-uswix162.iso'\'' to nfs location'
2016-03-23 12:44:34.902757177 Copying files '/var/lib/rear/output/rear-uswix162.iso' to nfs location
++ cp -v /var/lib/rear/output/rear-uswix162.iso /tmp/rear.yQCQmuw7D95ysgf/outputfs/uswix162/
cp: cannot stat `/var/lib/rear/output/rear-uswix162.iso': No such file or directory
++ StopIfError 'Could not copy files to nfs location'
++ ((  1 != 0  ))
++ Error 'Could not copy files to nfs location'
++ '[' Could not copy files to nfs location -eq Could not copy files to nfs location ']'
++ EXIT_CODE=1
++ VERBOSE=1
++ LogPrint 'ERROR: Could not copy files to nfs location'
++ Log 'ERROR: Could not copy files to nfs location'
++ test 1 -gt 0
+++ Stamp
+++ date '+%Y-%m-%d %H:%M:%S.%N '
++ echo '2016-03-23 12:44:34.907306449 ERROR: Could not copy files to nfs location'
2016-03-23 12:44:34.907306449 ERROR: Could not copy files to nfs location
++ Print 'ERROR: Could not copy files to nfs location'
++ test 1
++ echo -e 'ERROR: Could not copy files to nfs location'
++ has_binary caller
++ for bin in '$@'
++ type caller
++ return 0
++ echo '=== Stack trace ==='
=== Stack trace ===
++ local c=0
++ awk '
                                { l[NR]=$3":"$1" "$2 }
                                END { for (i=NR; i>0;) print "Trace "NR-i": "l[i--] }
                        '
++ caller 0
++ :
++ caller 1
++ :
++ caller 2
++ :
++ caller 3
++ :
++ caller 4
++ :
++ caller 5
++ :
++ caller 6
Trace 0: /usr/sbin/rear:249 main
Trace 1: /usr/share/rear/lib/mkbackup-workflow.sh:30 WORKFLOW_mkbackup
Trace 2: /usr/share/rear/lib/framework-functions.sh:79 SourceStage
Trace 3: /usr/share/rear/lib/framework-functions.sh:40 Source
Trace 4: /usr/share/rear/output/default/95_copy_result_files.sh:22 source
Trace 5: /usr/share/rear/lib/_input-output-functions.sh:132 StopIfError
++ echo 'Message: Could not copy files to nfs location'
Message: Could not copy files to nfs location
++ echo ===================


------------------Snipped---------------------

**Comment by gozora** [comment]
Hi,
This might be the problem: **Relax-and-Recover 1.16 / Git**
As far as I remember ebiso was first introduced in ReaR version 1.17 
@gdha correct me if I'm wrong

**Comment by parasharsingh** [comment]
That's correct. I just installed rear 1.18 on another system and set it up. That servers back is working fine.

Thanks for your help.

**Comment by gozora** [comment]
anytime ...

**Comment by gdha** [comment]
To be honest it was introduced 1.17.2-git-something, so in dev only. The 1st official support for ebiso is 1.18, hence the release.

**Comment by jsmeix** [comment]
As far as I see the above https://github.com/rear/rear/issues/804#issuecomment-200440869 is the same as https://github.com/rear/rear/issues/801

General information:

The newest rear and ebiso packages
for various openSUSE and SLE versions
are available from the openSUSE build service
for direct RPM download as RPM packages
named "rear-1.18-..." and "ebiso-0.2.1-..." from
<pre>
http://download.opensuse.org/repositories/Archiving:/Backup:/Rear/
</pre>
and (basically the same software) also from
<pre>
http://download.opensuse.org/repositories/Archiving/
</pre>

rear-1.18 includes several example config files in
<pre>
/usr/share/rear/conf/examples/
</pre>

In particular regarding UEFI, have a look at
<pre>
/usr/share/rear/conf/examples/SLE11-SLE12-SAP-HANA-UEFI-example.conf
</pre>

In general see
https://en.opensuse.org/SDB:Disaster_Recovery

As far as I understand https://github.com/rear/rear/issues/804#issuecomment-200469568 the issue can be closed because UEFI works with rear-1.18 on SLE when it is set up for ebiso.

For the future I am already thinking about if it is possible to automate it on SLE to use ebiso `if is_true $USING_UEFI_BOOTLOADER` (cf. https://github.com/rear/rear/issues/801) but that would be a separated new enhancement issue: https://github.com/rear/rear/issues/805
